config/config.go

```diff
@@ -32,10 +32,16 @@ func (cfg Config) String() string {
 	return string(b)
 }
 
-// New returns a default Config overridden with given parameters.
+// New returns a Config initialized with given parameters.
```

Comment:
Nitpick, we may want to document that is may be invalid/unusable until merged with `config.Merge`.